### PR TITLE
Only merge PR to 'preview' branch when 'preview' label is set and provide PR template with checklist (#695, #702, #744, #750, #330)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
# Description

EB Member: @bernhold

Addresses issues #695, #702, #744, #750, #330

This PR does two things.  First, it extends the `merge-pr-to-preview.yml` workflow to only merge PR branches to the 'preview' branch if the PR contains the `preview` label.   Second, it adds a PR template file for three types of PRs:

* a) bssw.io content
* b) bssw.io content with formal citations (using `wikize_refs.py`)
*c) non-bssw.io content

In installed this updated `merge-pr-to-preview.yml` workflow  to [my fork's](https://github.com/bartlettroscoe/bssw.io) 'master' branch and did some very detailed and carefully documented manual tests (see below). 

To try this out yourself, just create a branch off of the 'master' branch of [my fork](https://github.com/bartlettroscoe/bssw.io) and post a PR to that fork's 'master' branch and you will see the template in use.  Also, you can set the `preview` label and watch the PR's commits get merged (or not get merged) based on if that label is set or not.

## How was this tested?

I used the installation of these files in  [my fork's](https://github.com/bartlettroscoe/bssw.io) 'master' branch to test this out.  I created two PRs in my fork and pushed commits and set labels, etc., to test out the various use cases and carefully documented the test results.  The two test PRs were/are:

* https://github.com/bartlettroscoe/bssw.io/pull/7: **Add mention of 'preview' label on PR (#744, #330)**:  This is a PR changing files under the `docs/` directory not viewed on [bssw.io] and therefore no reason to set the `preview` label.  (This PR is never merged to the 'preview' branch in my fork).

* https://github.com/bartlettroscoe/bssw.io/pull/8: **Fix some whitespace**: This PR changes an `*.md` file that is viewed on the bssw.io site and I several events where I set or unset the `preview` label.

The tests I performed were:

* Reset 'rab-github/preview' back to 'rab-github/master' and force push (so 'preview' starts out idential to 'master') **[Done]**
* Test PR that does not get merged to 'preview' branch: **[Passed]**
  - Create new PR for branch '744-rab-preview-label-eb-docs-1' using the non-bssw.io template and don't set the 'preview' label (verfy that merge to 'preview' is skipped) **[Passed]**
  - Push another commit (verify that merge to 'preview' is skipped) **[Passed]**
* Test PR that should merge to 'preview' branch: **[Passed]**
  - Create new branch 'change-bssw-io-content-preview-label-1' with one commit **[Done]** 
  - Create new PR for branch 'change-bssw-io-content-preview-label-1' using bssw.io (verify job is skipped)  **[Passed]**
  - Add the label 'enhancement' (verify job is skipped) **[Passed]**
  - Add the 'preview' label (verify job is run merged to 'preview' branch) **[Passed]**
  - Remove the 'preview' label (verify no action occurs) **[Done]**
  - Push another commit to 'change-bssw-io-content-preview-label-1' (verify job is skipped because 'preview' label is not set) **[Passed]**
  - Add the 'preview' label (verify job is run merged to 'preview' branch) **[Passed]**
  - Push another commit to 'change-bssw-io-content-preview-label' (verify job is run merged to 'preview' branch) **[Passed]**

These manual tests are documented in more detail below 

<details>

<summary><b>Detailed manual test cases and evidence</b> (Click to expand)</summary>

.

**A) Create PR for files that will not to be merged to 'preview'**

I pushed the branch [744-rab-preview-label-eb-docs-1](https://github.com/bartlettroscoe/bssw.io/tree/744-rab-preview-label-eb-docs-1) that has the commit:

```
49beb658 "Add mention of 'preview' label on PR (#744, #330)"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Thu Mar 25 17:05:37 2021 -0400 (26 hours ago)

M       docs/pages/bssw/bssw_content_publishing.md
```

This mimics a change to files that will not be published in the bssw.io site so there is no reason to merge this branch to the 'preview' branch to be viewed on the preview.bssw.io site.

From that branch I create the following PR against the 'master' branch of my repo using the custom template:

* https://github.com/bartlettroscoe/bssw.io/pull/7

The creation of the PR triggered the `merge-pr-to-preview.yml` workflow which ran a skipped job:

* https://github.com/bartlettroscoe/bssw.io/pull/7/checks?check_run_id=2206155376

So GitHub Actions does not tell you why it was skipped which seems to be pretty sad.  I would have hopped that it would have printed out the condition that failed to explain this.

I then pushed the commit:

```
c08ba820 "DELETE ME: Add some vertial whitespace"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Thu Mar 25 17:36:11 2021 -0400 (26 hours ago)

M       docs/pages/bssw/bssw_content_publishing.md
```

and that triggered another invocation of the `merge-pr-to-preview.yml` workflow and produced the skipped job:

* https://github.com/bartlettroscoe/bssw.io/pull/7/checks?check_run_id=2206184385

And fetching the updated branches, the 'preview' branch is unchanged compared to 'master' as shown by:

```
$ git fetch rab-github

$ git log-short --name-status rab-github/preview --not rab-github/master
[empty]
```

That that test shows that by not adding the 'preview' label, the PR branch does **not** get merged to the 'preview' label.

[PASSED]

**B) Create a test a PR for content that will be viewed on bssw.io**

For this use case, I created the branch 'change-content-file-test-pr-preview-label-1' that pushes updates to the file:

```
CuratedContent/WorkingEffectivelyWithLegacyCode.md
```

in several commits.

I pushed the first commit:

```
b778f2d7 "Fix some whitespace"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Thu Mar 25 18:08:18 2021 -0400 (26 hours ago)

M       CuratedContent/WorkingEffectivelyWithLegacyCode.md
```

to the branch [change-content-file-test-pr-preview-label-1](https://github.com/bartlettroscoe/bssw.io/tree/change-content-file-test-pr-preview-label-1) for which I created the initial PR:

* https://github.com/bartlettroscoe/bssw.io/pull/8

You can see that the creation of this PR fired off the `merge-pr-to-preview.yml` workflow and produced the skiped job:

* https://github.com/bartlettroscoe/bssw.io/actions/runs/691712283

I then added the label `enhancement` but I did **not** set the label `preveiw`.  That fired off the `merge-pr-to-preview.yml` workflow again and again produced a skiped job:

* https://github.com/bartlettroscoe/bssw.io/actions/runs/691712284

As of this point, the 'preview' branch has **not** been updated as shown by:

```
$ git fetch rab-github 

$ git log-short --name-status rab-github/preview --not rab-github/master
[empty]
```

Next, I added the `preview` label to that PR, without making any other changes or pushing any new commits and that triggered the `merge-pr-to-preview.yml` workflow which produced the non-skipped job:

* https://github.com/bartlettroscoe/bssw.io/actions/runs/691719762

which took 24s to run and produced the updated 'preview' branch:

```
$ git fetch rab-github
...
From github.com:bartlettroscoe/bssw.io
   342a38c6..758f3b2c  preview    -> rab-github/preview

$ git log-short --name-status rab-github/preview --not rab-github/master 

758f3b2c "Merge b778f2d74b8037f7dcc4a148208748db3856b24a into preview"
Author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Sat Mar 27 00:57:50 2021 +0000 (36 seconds ago)

b778f2d7 "Fix some whitespace"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Thu Mar 25 18:08:18 2021 -0400 (27 hours ago)

M       CuratedContent/WorkingEffectivelyWithLegacyCode.md
```

So that worked great!

Next, I removed the `preview` label from the PR.  That did **not** trigger the `merge-pr-to-preview.yml` workflow because that workflow does not have the `unlabled` type for the `pull_request_target` event.  (We don't want to do anything if a label is removed so this is good.)

Next, I pushed the commit:

```
1ba4c4b0 "Change from - to * for some bullets"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Thu Mar 25 18:21:24 2021 -0400 (27 hours ago)

M       CuratedContent/WorkingEffectivelyWithLegacyCode.md
```

That triggered the `merge-pr-to-preview.yml` workflow but produced as skipped job:

* https://github.com/bartlettroscoe/bssw.io/actions/runs/691730004

That is what we want because we removed the `preview` label which means that we don't want to be merging to the 'preview' branch for now.

But then I set the `preview` label for that PR without changing anything else and it triggered the `merge-pr-to-preview.yml` workflow again with the job:

* https://github.com/bartlettroscoe/bssw.io/actions/runs/691734090

which took 27s which merged the new commit an create the new commits on the 'preview' branch:

```
$ git fetch rab-github
...
From github.com:bartlettroscoe/bssw.io
   758f3b2c..f65588e6  preview    -> rab-github/preview

$ git log-short --name-status rab-github/preview --not 758f3b2c

f65588e6 "Merge 1ba4c4b02ed7004177dadddde859c970ac595063 into preview"
Author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Sat Mar 27 01:06:38 2021 +0000 (35 seconds ago)

1ba4c4b0 "Change from - to * for some bullets"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Thu Mar 25 18:21:24 2021 -0400 (27 hours ago)

M       CuratedContent/WorkingEffectivelyWithLegacyCode.md
```

Finally, I pushed the commit:

```
1f43ffa0 "More Whitespace changes"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Thu Mar 25 18:45:48 2021 -0400 (26 hours ago)

M       CuratedContent/WorkingEffectivelyWithLegacyCode.md
```

and the PR which already has the `preview` label set which triggered the `merge-pr-to-preview.yml` workflow again which ran the job:

* https://github.com/bartlettroscoe/bssw.io/actions/runs/691739198

which took 35s to complete and produced the new commits on the 'preview' branch:

```
$ git fetch rab-github
...
From github.com:bartlettroscoe/bssw.io
   f65588e6..c0257920  preview    -> rab-github/preview

$ git log-short --name-status rab-github/preview --not f65588e6
c0257920 "Merge 1f43ffa0525ff68b431dcf3c9d281680718683b8 into preview"
Author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Sat Mar 27 01:10:02 2021 +0000 (70 seconds ago)

1f43ffa0 "More Whitespace changes"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Thu Mar 25 18:45:48 2021 -0400 (26 hours ago)

M       CuratedContent/WorkingEffectivelyWithLegacyCode.md
```

That is pretty good evidence that this this `merge-pr-to-preview.yml` workflow behaves as intended.  It also shows that it works even when multiple labels are set.  NOTE: For multiple labels, you have to use the conditional:

```
if: contains(github.event.pull_request.labels.*.name, 'preview')
```

and **not** the conditional:

```
if: ${{ github.event.label.name == 'preview' }}
```

That latter fails when more than one label is set.  (I verified this during earlier development.)

</details>

## Checklist for (internal) files not displayed on bssw.io site

* [x] Set list of Reviewers (please at least one).
* [x] Add to Project `BSSw Internal`.
* [x] View the modified `*.md` files as rendered in GitHub.
* [ ] If changes are to the GitHub pages site under the `docs/` directory, consider viewing locally with Jekyll.
* [ ] Watch for PR check failures.
* [ ] Make any final changes to the PR based on feedback and review GitHub (and Jekyll) rendered files.
* [ ] Ensure at least one reviewer signs off on the changes.
* [ ] Once reviewer has approved and PR check pass, then merge the PR.

[preview]: https://preview.bssw.io
[bssw.io]: https://bssw.io
[content development]: https://github.com/betterscientificsoftware/bssw.io/projects/3?
[meta-data]: https://betterscientificsoftware.github.io/bssw.io/bssw_styling_common.html#metadata-section
[wikize_refs.py]: https://github.com/betterscientificsoftware/bssw.io/blob/master/utils/README.md#wikize_refspy

### CODE REVIEW

**Comment by bartlettroscoe** [comment]
@bernhold, I resolved the conflict from 'master' to 'preview' in the commit https://github.com/betterscientificsoftware/bssw.io/commit/8c371e953e890290785d95c7faef66ad515f931b that was blocking this PR from merging to 'preview' (which the current workflow `merge-pr-to-preview.yml` does unconditionally which this PR fixes).  But now it is failing as shown at:

* https://github.com/betterscientificsoftware/bssw.io/runs/2209196762?check_suite_focus=true

with:

> /usr/local/bundle/gems/octokit-4.14.0/lib/octokit/response/raise_error.rb:16:in `on_complete': POST https://api.github.com/repos/betterscientificsoftware/bssw.io/merges: 403 - Resource not accessible by integration // See: https://docs.github.com/rest/reference/repos#merge-a-branch (Octokit::Forbidden)

Do you have any idea what might be causing this?  I don't see any changes in permissions on this bssw.io GH repo that might be causing this.

**Comment by bartlettroscoe** [comment]
Some comments about using a single PR template file `.github/PULL_REQUEST_TEMPLATE.md` vs. a set of different template files is in order.  As explained in:

* https://blog.axosoft.com/enhancing-pull-request-descriptions-templates/

GitHub supports multiple PR template files but as I note in https://github.com/betterscientificsoftware/bssw.io/pull/765/commits/0756e247262781c9955ede1716d4b50777ec1db9 the only way to use them is to **manually** edit the create-PR URL to add `template=<template-name>` to the end of the URL.  That is completely unworkable in my opinion.  Therefore, I chose to create a single default PR template that can be quickly edited to select one of three different checklists for different types of PRs.

For Issue templates, it is a completely different story.  As shown with the bssw.io Issue templates shown [here](https://github.com/betterscientificsoftware/bssw.io/issues/new/choose), GitHub displays a nice page allowing you to select from a list of templates.  The issue is described very well here:

* https://github.community/t/multiple-pull-request-templates/1850

I added my two-cents there as well:

* https://github.community/t/multiple-pull-request-templates/1850/13?u=bartlettroscoe

But we work with what we have got and I think the meta-template approach that I have implemented in this PR is about the best we can do I think.

**Comment by bartlettroscoe** [comment]
@bernhold, responding to your comment in https://github.com/betterscientificsoftware/bssw.io/issues/695#issuecomment-808771709 here since it is specific to this PR.

>  I would modify the end of your checklist

I updated with your suggested checklist items in https://github.com/betterscientificsoftware/bssw.io/pull/765/commits/7e73676874d7f222abef8ba1410419e4f54e2bc4 pushed to this PR.  Is there anything else you see that needs to be done before we merge this?

> The "site maintainers" bit needs to be better defined. What we have now are a subset of the BSSw Team who know the process to login and rebuild the production site. In theory we could open this up to the whole EB, though being the production site, we've kept it to a smaller group of people to date.

At some point, the update of the bssw.io and preview.bssw.io sites really needs to be automated for this to be sustainable.  A simple local polling of the GitHub repo with `git fetch ...` in a loop should do it.  Just to many manual steps to this process right now.

Another thing that occurred to me is that it might help to break down the checklist into different subsections for the **Author** and the **EB Member** to perform.  The current checklist does not make it clear what checklist item should be done by which role.  And we should really minimize that items we expect the Author to do and push as much as we can on the EB Member to make this process less intimidating for authors.   (But Authors need to own the content and should be expected to make changes to the content based on review feedback.  Issues with translation and viewing on the preview.bssw.io site should be take care of by the EB Member.   So the EB Member should be setting the 'preview' label and update the preview.bssw.io site, not the Author in most cases.) We might also consider a set of checklist items for a **Reviewer** role.  But we should not get too crazy with checklist items because the "bssw.io files with formal citations" checklist is already quite long with 22 items (all of which have to be done)!

**Review by bernhold — APPROVED** [review]
I think the template could be improved, but the perfect is the enemy of the good, and this is something we can work with.

The workflow change looks good too.  The way you've got it set, the action would be triggered when a PR is opened or updated *or* when a label is applied to the PR.  But triggering only on the labeling would not give us the results we want either (we wouldn't update preview when the PR is updated).  I think your solution is good.  Sorry for misleading you.

**Comment by markcmiller86** [comment]
> Therefore, I chose to create a single default PR template that can be quickly edited to select one of three different checklists for different types of PRs.

I haven't reviewed this PR in detail but it seems to be a *single* PR template for outward facing authors only with a checklist that has sublists would work ok as in...

* [ ] `@mention` the BSSw.io editorial board member `@<eb-member-id>` in **Description** above assigned to shepherd your PR.
* [ ] Add the `<issue-id>` in the **Description** above for the associated GitHub Issue.
* [ ] Assign this PR to the EB member `<eb-member-id>`.
* [ ] Assign this PR to the author of the PR `<pr-author-id>`.
* [ ] Add label `content: <content-type>` for the type of contribution.
* [ ] Add to Project `Content Development` (see [content development]).
* [ ] Inspect the content in the `*.md` file(s) as rendered in GitHub for this PR.
* [ ] Add one or more Reviewers.
* [ ] Add [meta-data] to the `*.md` file(s) (set `Publish: preview`).
* [ ] Add label `preview` (so PR branch will be merged to 'preview' branch and watch for possible merge failures).
   * [ ] If using citations, ensure `wikize_refs.py -i <base>.md` is run and commit (see [wikize_refs.py]) otherwise bracket this line in strikeout (`~~`)
* [ ] Rebuild [preview] site and confirm new content is there, renders correctly and is returned in searches.
* [ ] Make any final changes to the PR based on feedback.
   * [ ] If using citations, ensure `wikize_refs.py -i <base>.md` is run and commit (see [wikize_refs.py]) otherwise bracket this line in strikeout (`~~`)
* [ ] Rebuild [preview] site and re-confirm content looks correct.
* [ ] Ensure at least one reviewer signs off on the final changes.
* [ ] Change meta-data to `Publish: yes` and commit if fully ready to publish.
* [ ] Move the PR to "Ready to Publish" in [content development].
* [ ] Assign to one of the site maintainers to carry out final publication steps.
* [ ] Verify that all needed files are present in the PR (article, images, updates to Site/Homepage.md carousel and/or Site/Announcements/Announcements.md as appropriate).
* [ ] Merge PR. Should automatically move to "Done" in [content development].
* [ ] Verify new new contribution shows up on [bssw.io] as expected.

If the PR is something EB members are doing, then that checlist is elsewhere and not part of the PR template.

**Comment by bernhold** [comment]
> * https://github.com/betterscientificsoftware/bssw.io/runs/2209196762?check_suite_focus=true
> 
> with:
> 
> > /usr/local/bundle/gems/octokit-4.14.0/lib/octokit/response/raise_error.rb:16:in `on_complete': POST https://api.github.com/repos/betterscientificsoftware/bssw.io/merges: 403 - Resource not accessible by integration // See: https://docs.github.com/rest/reference/repos#merge-a-branch (Octokit::Forbidden)
> 
> Do you have any idea what might be causing this? I don't see any changes in permissions on this bssw.io GH repo that might be causing this.

@bartlettroscoe I think this is some kind of permissions or access control thing.  There are nuances when the PR comes from a fork outside of the repo.  But I've never fully come to grips with them.  I'm pretty sure I've tested cases like this from forks and they were okay, but I did have trouble with other automations I was trying to use when operating from a fork.

I bet if you did this from a branch in bssw.io, it would be fine, but, as above, I don't understand why yet.

**Comment by markcmiller86** [comment]
FYI, we have a similar issue with several checklist items that might be `N/A` in VisIt. So, we pre-build that checklist to facilitate quick strikeout of `N/A` items as in...

https://raw.githubusercontent.com/visit-dav/visit/develop/.github/pull_request_template.md

**Comment by markcmiller86** [comment]
So, we can wind up with PR comments that look like this...

https://github.com/visit-dav/visit/pull/5557

**Comment by bernhold** [comment]
@markcmiller86 

> [visit-dav/visit#5557](https://github.com/visit-dav/visit/pull/5557)

I can't see the raw markup, but I take it that you add ~~ in front of items that are not applicable and with the trailing ~~ that strikes out the item?

**Comment by markcmiller86** [comment]
> I can't see the raw markup, but I take it that you add ~~ in front of items that are not applicable and with the trailing ~~ that strikes out the item?

Ooops. Sorry. There is no `Raw` view option there and `...`->`Edit` is only available to *members*. Pasting raw below...

```
- [X] I have followed the [style guidelines][1] of this project.~~
- [X] I have performed a self-review of my own code.~~
~~- [ ] I have commented my code where applicable.~~
~~- [ ] I have updated the release notes.~~
~~- [ ] I have made corresponding changes to the documentation.~~
~~- [ ] I have added debugging support to my changes.~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works.~~
~~- [ ] I have confirmed new and existing unit tests pass locally with my changes.~~
~~- [ ] I have added any new baselines to the repo.~~
~~- [ ] I have NOT made any changes to [*protocol* or *public interfaces*][3] in an RC branch.~~
~~- [ ] I have assigned reviewers (see [VisIt's PR procedures][2] for more information).~~

[1]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/StyleGuide.html
[2]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/pr_create.html#reviewers
[3]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/RCDevelopment.html#communication-protocols-and-public-apis
```

In the template, we *end* each line in the PR with `~~` which by itself looks odd but is otherwise harmless. Then, to fully strikeout a line, a submitter need only add `~~` at the beginning of the relevant lines.

**Comment by bartlettroscoe** [comment]
> I haven't reviewed this PR in detail but it seems to be a single PR template for outward facing authors only with a checklist that has sublists would work ok as in...

@markcmiller86, I had considered that but was not sure how people feel about that.  It would be easier to maintain this checklist if it was just one combined list.  @bernhold, are you okay with just one list with optional sublists like this?

> In the template, we *end* each line in the PR template with `~~` which by itself looks odd but is otherwise harmless. Then, to fully strikeout a line, a submitter need only add `~~` at the beginning of the relevant lines.

@markcmiller86, that looks fine to me.  We can even mention the usage of `~~` to strike out lines for checklist items that don't apply in the **Instructions** section at the top of the PR template.

BTW, I edited the [above comment](https://github.com/betterscientificsoftware/bssw.io/pull/765#issuecomment-808796845) to show what the rendered Markdown of the visit checklist looks like.

I will implement these changes and push them to the template.

**Comment by bartlettroscoe** [comment]
>There are nuances when the PR comes from a fork outside of the repo.

@bernhold, while I can push branches to the bssw.io repo, outside contributors can't.  Therefore, we should figure out if that is the problem or not.  It does not matter for this PR since this is not content we need to view on the preview.bssw.io site but we should keep an eye on this for future PRs.  I fact, after we merge this PR, I will post a CC article and work our the issues in that PR.

**Comment by markcmiller86** [comment]
> I will implement these changes and push them to the template.

FYI...I think it looks nicer (and for that reason, I kinda sorta prefer), that when an item in checklist **does not apply**, it is striken out by adding the `~~` to either *replace* the `[ ]` or *just* ahead of the `[ ]` and not before the bullet as in...

- [ ] this item is not striken out
- ~~this item is striken out by replacing `[ ]` with `~~`~~
- ~~[ ] this item is striken out by putting `~~` just ahead of `[ ]`~~
~~- [ ] this item is striken out by placing `~~` ahead of bullet itself~~

**Comment by bartlettroscoe** [comment]
@markcmiller86, thanks for the excellent feedback.  I pushed commit https://github.com/betterscientificsoftware/bssw.io/pull/765/commits/72215726acb6c4e99ca87def33644dd774f66c1c that should implement your advice.

I think this is ready to merge.  We can make more tweaks to the checklists as follow-on PRs.

**Comment by bartlettroscoe** [comment]
FYI: After the merge of this PR I noticed that the checklist item to add the `preview` label was **after** the checklist item to rebuild the preview.bssw.io site.  Since the checklist items should be in order, I pushed the commit https://github.com/betterscientificsoftware/bssw.io/commit/a067ceef7e6485c4a632fe5dbb450e2f836d7858 to fix that.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (markcmiller86) [inline .github/PULL_REQUEST_TEMPLATE.md:50]: Ok so it looks like the checklist is setup to *default* strikeout those items related to citations/refs. That seems like a good idea because that use case is less common.

The note at the bottom then confuses me a bit because it describes how to *stikeout* (more) optional items and not how to **UN**strikeout items that may need to apply (e.g. for cases where citations/refs are used).

Comment (bartlettroscoe) [reply .github/PULL_REQUEST_TEMPLATE.md:50]: > The note at the bottom then confuses me a bit because it describes how to stikeout (more) optional items and not how to UNstrikeout items that may need to apply (e.g. for cases where citations/refs are used).

@markcmiller86, the instructions above mention removing strikeout markers for the wikize_refs.py item but it is worth mentioning removing strikeout markers in this below note as well.  I can change that sentence to mention adding and remove strikeout markers `~~`.  I will post that as a new PR so we can iterate more on this checklist now that we have something in place.

Comment (bartlettroscoe) [reply .github/PULL_REQUEST_TEMPLATE.md:50]: @markcmiller86, I posted PR #771 with that change.  Please provide comments in that PR #771 for any other thing you might see that needs tweaked.

**Pull request merged**
